### ISSUE TITLE
"Last region" doesn't work with multiple monitors

### ISSUE THREAD

**Issue by OddMorning** [ShareX/ShareX#6624]
Recently I moved my second monitor to the left side of the main one and I noticed that the "Last region" feature doesn't work properly anymore.

For example, when I take a screenshot of the green area the normal way, the "last region" takes a screenshot of the red area:
![pic](https://user-images.githubusercontent.com/12204657/205254818-477a130f-b9f4-4dd5-840a-07d7b9722630.png)

**Comment by StelioKontosXBL** [comment]
- Is the primary display the same as it was before?
- Have you restarted your PC, or at least restarted the ShareX application since changing the setup?

**Comment by OddMorning** [comment]
Just did more tests (restarts & settings changes) and it has nothing to do with a primary/secondary status. It seems "last region" stores relative monitor coordinates but later threats them as absolute screen coordinates.

This monitors placement results to nothing but black images as there are no monitors in the top left corner:
![ApplicationFrameHost – 03 12 2022; 14 36 44](https://user-images.githubusercontent.com/12204657/205439702-45ca6e50-9ab6-4d40-93dc-7b4122116548.png)

And this attempt to take a screen 1 takes a half of screen 2:
![ApplicationFrameHost – 03 12 2022; 14 36 441](https://user-images.githubusercontent.com/12204657/205439708-b96982a9-eaa3-4e43-8e80-f808e2f0ad36.png)

**Comment by StelioKontosXBL** [comment]
I attempted to repro what you have described yesterday with two 32-inch displays, both running at 2560 x 1440. The only way I was able to get the same result you are describing was by rearranging the monitors *between* taking the initial screenshot and using the "Last Region" option (and thereby invalidating the screenshot coordinates cached when the initial screenshot was taken).

If I switched the two monitors, took a screenshot, then used the "Last Region" option, it worked without fail.
